integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/PathUtils.java

```diff
@@ -157,6 +159,11 @@ private static Optional<String> getGlueArn(
 
     Optional<String> clientFactory =
         SparkConfUtils.findHadoopConfigKey(hadoopConf, "hive.metastore.client.factory.class");
+    // Fetch from spark config if set.
+    clientFactory =
+        clientFactory.isPresent()
+            ? clientFactory
+            : SparkConfUtils.findSparkConfigKey(sparkConf, "hive.metastore.client.factory.class");
```

Comment:
But Spark config properties do not start with `hive.`

Reply:
We can pass the hive configs to Spark configs as well.  Atleast for these Glue catalog config. 
eg: 
```
spark = SparkSession.builder.master("yarn").appName(f'{app_name}').\
            config("spark.shuffle.blockTransferService", "nio").\
            config("spark.sql.parquet.enableVectorizedReader", "false"). \
            config("spark.sql.sources.partitionOverwriteMode", "dynamic"). \
            config("spark.sql.sources.parallelPartitionDiscovery.parallelism", "10"). \
            config(conf=spark_conf).\
            config("hive.metastore.client.factory.class", "com.amazonaws.glue.catalog.metastore.AWSGlueDataCatalogHiveClientFactory").\
            config("hive.metastore.glue.catalogid",catalog_id).enableHiveSupport().getOrCreate()
```

The same is mentioned in EMR guide as well: https://docs.aws.amazon.com/emr/latest/ReleaseGuide/emr-spark-glue.html

Tested this with an actual EMR glue catalog job.

Reply:
Hm, never heard about this, okay